### PR TITLE
Enterprise release notes page

### PR DESCRIPTION
**Pull request by victoriaspek**
Created a page in the docs to keep a running list of Enterprise release notes. Let me know what you think!

### CODE REVIEW

File: docs/enterprise-guide/start.md [hunk 1]
```diff
@@ -10,3 +10,6 @@ The Enterprise Edition of Metabase provides added functionality and solutions fo
 * [Customizing how Metabase looks with white labeling](whitelabeling.md)
 * [Customizing drill-through for charts and tables](customizing-drill-through.md)
 * [Copying contents of one Metabase instance to another (serialization)](serialization.md)
+
+
+[Click here](release-notes.md) to find release notes for our latest version of Metabase Enterprise.
```
Comment (mazameli) [inline docs/enterprise-guide/start.md]: I'd favor `Check out the [release notes](release-notes.md) for the latest version of the Enterprise Edition.` Personal pet peeve against "Click here" links. :b

**Review by mazameli — COMMENTED** [review]
Left two small notes. Only thing that's kind of a bummer is that updating this page will cause a bit of manual drag until/unless we're able to automatically update this page with the release notes.

Another sad fact is that publishing updates to this page will depend on new versions of the _open source_ version getting shipped because of how our docs are currently deployed (unless we do these updates over in the website repo as well). My understanding of why we've historically coupled docs with OSS releases is so that we can have versioned docs. But maybe it's time to decouple docs from OSS releases, or at least somehow decouple enterprise docs?

**Comment by mbvictoria** [comment]
Yeah, I think that it probably makes sense to rethink the docs system sometime in the near future, especially since it looks like we're going to be creating guides that may be completely separate from the release cycle. 

For now, I can update the content for this page, and we can reuse it as the content for the email, I don't mind.

If we end up having the email to notify people of a new version, I think it's okay for this to be a little less real-time as well.

**Comment by salsakran** [comment]
I don't think this belongs in this repository.

Can we move it to the website repository?

**Comment by mbvictoria** [comment]
Closing - this ended up on the website.

**Pull request closed without merging**